lib/teiserver/microblog/libs/poll_response_lib.ex

```diff
@@ -0,0 +1,163 @@
+defmodule Teiserver.Microblog.PollResponseLib do
+  @moduledoc false
+  use TeiserverWeb, :library_newform
+  alias Teiserver.Microblog.{PollResponse, PollResponseQueries}
+  alias Phoenix.PubSub
+
+  @doc """
+  Returns the list of poll_responses.
+
+  ## Examples
+
+      iex> list_poll_responses()
+      [%PollResponse{}, ...]
+
+  """
+  @spec list_poll_responses(list) :: [PollResponse]
+  def list_poll_responses(args \\ []) do
+    args
+    |> PollResponseQueries.query_poll_responses()
+    |> Repo.all()
+  end
+
+  @spec get_poll_response(non_neg_integer(), non_neg_integer()) ::
+          PollResponse | nil
+  def get_poll_response(user_id, post_id) do
+    PollResponseQueries.query_poll_responses(
+      where: [
+        user_id: user_id,
+        post_id: post_id
+      ],
+      limit: 1
+    )
+    |> Repo.one()
+  end
+
+  @doc """
+  Creates a poll_response.
+
+  ## Examples
+
+      iex> create_poll_response(%{field: value})
+      {:ok, %PollResponse{}}
+
+      iex> create_poll_response(%{field: bad_value})
+      {:error, %Ecto.Changeset{}}
+
+  """
+  def create_poll_response(attrs \\ %{}) do
+    %PollResponse{}
+    |> PollResponse.changeset(attrs)
+    |> Repo.insert()
+    |> broadcast_create_poll_response
+  end
+
+  defp broadcast_create_poll_response({:ok, %PollResponse{} = poll_response}) do
+    spawn(fn ->
+      # We sleep this because sometimes the message is seen fast enough the database doesn't
+      # show as having the new data (row lock maybe?)
+      :timer.sleep(1000)
+
+      PubSub.broadcast(
+        Teiserver.PubSub,
+        "microblog_poll_responses",
+        %{
+          channel: "microblog_poll_responses",
```

Comment:
Where is this subscribed to? I didn't find anything grepping through the code.

Reply:
It's not but it's part of the way I'd used other things and wanted to provide an option for it to be subscribed to. Happy to remove it if you're not happy with it.